File: McMaster-Week-of-Code/Week-1/Staircase.cpp
#include <bits/stdc++.h>

using namespace std;

// Complete the staircase function below.
void staircase(int n) {
    char output[100];
    
    for (int i = 0; i < n; ++i) {
        output[i] = ' ';
    }
    output[n] = 0;
    
    for (int i = n-1; i >= 0; --i) {
        output[i] = '#';
        cout << output << endl;
    }
}

int main()
{
    int n;
    cin >> n;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    staircase(n);

    return 0;
}

File: McMaster-Week-of-Code/Week-1/Staircase.py
#!/bin/python

import math
import os
import random
import re
import sys

# Complete the staircase function below.
def staircase(n):
    output = [' ']*n
    for i in range(n-1, -1 , -1):
        output[i] = '#'
        print ''.join(output)

if __name__ == '__main__':
    n = int(raw_input())

    staircase(n)
